controllers/complianceeventsapi/server.go

```diff
@@ -471,6 +505,93 @@ func parseQueryArgs(queryArgs url.Values, isCSV bool) (*queryOptions, error) {
 		}
 	}
 
+	parsed, err := setAuthorizedClusters(ctx, db, parsed, userConfig)
+	if err != nil {
+		// ErrNoAccess needs queryOptions
+		return parsed, err
+	}
+
+	return parsed, nil
+}
+
+// getAuthorizedClusterNames verifies that if a cluster filter is provided,
+// the user has access to this filter. If no cluster filter is provided,
+// it sets the cluster filter to all managed clusters the user has access to.
+func setAuthorizedClusters(ctx context.Context, db *sql.DB, parsed *queryOptions,
+	userConfig *rest.Config,
+) (*queryOptions, error) {
+	unAuthorizedClusters := []string{}
+
+	// Get all managedCluster rules
+	allRules, err := getManagedClusterRules(userConfig, nil)
+	if err != nil {
+		return parsed, err
+	}
+
+	tempClusterIDs, okClusterIds := parsed.Filters["clusters.cluster_id"]
+	clusterIDs := make([]string, len(tempClusterIDs))
+
+	// Deep copy parsed.Filters["clusters.cluster_id"]
+	copy(clusterIDs, tempClusterIDs)
+
+	if okClusterIds {
+		// Convert id to name
+		for index, id := range clusterIDs {
+			clusterName, err := getClusterNameFromID(ctx, db, id)
+			if err != nil {
+				if errors.Is(err, sql.ErrNoRows) {
+					// This leads to send forbidden error to the user
+					parsed.Filters["clusters.cluster_id"] = append(parsed.Filters["clusters.cluster_id"][:index],
```

Comment:
This wouldn't work if there was more than one invalid cluster ID since index no longer matches. My suggestion is traverse `clusterIDs` in reverse so that removing any cluster ID does not impact the current index.

Reply:
isn't better we reset parsed.Filters["clusters.cluster_id"] then append?